### PR TITLE
WFLY-10222 Don't propagate exception if reading the resource fails

### PR DESCRIPTION
**Pull request by stuartwdouglas**
https://issues.jboss.org/browse/WFLY-10222

### CODE REVIEW

**Comment by stuartwdouglas** [comment]
Retest this please

File: jaxrs/src/main/java/org/jboss/as/jaxrs/DeploymentRestResourcesDefintion.java [hunk 1]
```diff
@@ -179,60 +184,65 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
...
 
-                    return null;
+                        return null;
+                    }
                 }).call(null, null);
 
             } catch (Exception ex) {
-                throw new RuntimeException(ex);
+                //WFLY-10222 we don't want a failure to read the attribute to break everything
+                JaxrsLogger.JAXRS_LOGGER.failedToReadAttribute(ex);
```
Comment (bstansberry) [inline jaxrs/src/main/java/org/jboss/as/jaxrs/DeploymentRestResourcesDefintion.java]: This can report both the address and the attribute name.

**Comment by stuartwdouglas** [comment]
Retest this please

**Comment by stuartwdouglas** [comment]
@bstansberry does this look ok now? In particular is the use of context.addResponseWarning an appropriate way of dealing with this situation?

**Comment by bstansberry** [comment]
@stuartwdouglas I think this is ok if it is a critical issue. I'd like to hear your thoughts on https://issues.jboss.org/browse/WFCORE-3754 though.

@jmesnil FYI.

**Comment by stuartwdouglas** [comment]
It is not a critical issue, if there are plans to solve this in a more general fashion then this can wait.

**Comment by kabir** [comment]
@stuartwdouglas @bstansberry adding the hold label until you decide what to do with this

**Comment by bstansberry** [comment]
@stuartwdouglas  I think it's fine to merge this, so long as on the JIRA there's a big obvious comment saying that this is a workaround fix and it's not a regression if we do WFCORE-3754 and then remove this, breaking people who don't set the param discussed in WFCORE-3754. Expectation would be HAL would set the param. I believe HAL is the typical use case for reading an entire deployment resource tree and hitting this problem.

I have no idea when we'd get to WFCORE-3754 so it seems silly to leave people broken.

**Review by bstansberry — APPROVED** [review]
I added the comment I wanted to the JIRA. This is ok with me so I removed the hold label.

**Pull request merged**